circuits/lib/utils/spongeHash.circom

```diff
@@ -0,0 +1,59 @@
+pragma circom 2.0.0;
+
+include "../../../node_modules/circomlib/circuits/poseidon.circom";
+
+// Because of the way the Poseidon hash function is implemented for Solidity, the max number of inputs must be 6
+template SpongeHash(arraySize) {
+	signal input in[arraySize];
+	signal output out;
+    // batch size is 5 because 1 input is reserved for the hash of the previous iteration
+    var batchSize = 5;
+    var iterationCount = 0;
+	var hashFnBatchSize = 6;
+    component firstPoseidon = Poseidon(hashFnBatchSize);
+    for(var i = 0; i < hashFnBatchSize; i++) {
+        firstPoseidon.inputs[i] <== getArrayValueByIndex(in, arraySize, i);
+    }
+
+    var restLength = arraySize - hashFnBatchSize;
+	if (restLength > batchSize) {
+		var r = restLength % batchSize;
+		var diff = r == 0 ? 0 : batchSize - r;
+		iterationCount = (restLength + diff) / batchSize;
+	}
```

Comment:
Something with the tabulation

Reply:
everything is fine in my editor
<img width="1101" alt="image" src="https://user-images.githubusercontent.com/23165148/216049854-4253f089-7c04-4a98-9097-c5dd5aa3d574.png">